### PR TITLE
Adding Workflow Management

### PR DESCRIPTION
**Pull request by gordonkoehn**
Implementation of reproducible and scalable workflows using `snakemake`. Currently, only implements the mock data generation using `cell_simulation.py`.

The code structure and environmental setup are described in the `workflows/README.md`

Looking forward to your feedback and thoughts on the setup!

### CODE REVIEW

File: scripts/cell_simulation.py [hunk 1]
```diff
@@ -67,7 +94,8 @@ def create_parser() -> argparse.Namespace:
         "--observe_homozygous",
         required=True,
         help="Observing homozygous mutations",
-        type=bool,
+        choices=[False, True],
+        type=t_or_f,
```
Comment (gordonkoehn) [inline scripts/cell_simulation.py:98]: There was a bug here. `cell_simulation.py` did not parse a string "False" correctly. It always parsed `True`. Hence, this custom function `t_or_f`. Please let me know if there is a more elegant way.

Comment (pawel-czyz) [reply scripts/cell_simulation.py:98]: Actually, I'd recommend not making it required and setting it to `False` by default.

This should work:
```
parser.add_argument(
    "--observe-homozygous",
    help="Observing homozygous mutations. By default False.",
    action="store_true",
)
```

**Comment by pawel-czyz** [comment]
Very nice! Have you had prior experience with Snakemake? It's cool that you know how to use `expand` and wildcards :slightly_smiling_face:

**Comment by gordonkoehn** [comment]
Great. Thanks for the Review! - hope to learn Snakemake properly further down the line.

**Pull request merged**